src/imageio/JxlImageLoader.cpp

```diff
@@ -0,0 +1,233 @@
+            status = JxlDecoderProcessInput(decoder.get());
+        } else if (status == JXL_DEC_COLOR_ENCODING) {
+            // Get color profile information
+            JxlColorEncoding color_encoding;
+            if (JXL_DEC_SUCCESS == JxlDecoderGetColorAsEncodedProfile(decoder.get(), JXL_COLOR_PROFILE_TARGET_DATA, &color_encoding)) {
+                // If the image is not already in sRGB/Rec709, we may need to convert
+                if (color_encoding.white_point != JXL_WHITE_POINT_D65 || 
+                    color_encoding.primaries != JXL_PRIMARIES_SRGB) {
+                    // This is a simplified approach - for a full implementation 
+                    // we'd need proper color space conversion based on the specific primaries
+                    if (color_encoding.primaries == JXL_PRIMARIES_CUSTOM) {
+                        // Get custom primaries if available
+                        float primaries[8];
+                        size_t primaries_size = sizeof(primaries);
+                        if (JXL_DEC_SUCCESS == JxlDecoderGetColorAsICCProfile(decoder.get(), JXL_COLOR_PROFILE_TARGET_ORIGINAL, nullptr, primaries_size)) {
+                            // Set up conversion matrix - simplified approach
+                            // In a real implementation, we'd need to properly extract the primaries
+                            // from the ICC profile and calculate an accurate conversion matrix
+                            toRec709 = Matrix4f(1.0); // Placeholder
+                        }
+                    }
+                }
+            }
+            
+            status = JxlDecoderProcessInput(decoder.get());
+        } else if (status == JXL_DEC_NEED_IMAGE_OUT_BUFFER) {
+            size_t buffer_size;
+            if (JXL_DEC_SUCCESS != JxlDecoderImageOutBufferSize(decoder.get(), &format, &buffer_size)) {
+                throw runtime_error{"Failed to get output buffer size for JPEG XL image."};
+            }
+
+            pixels.resize(buffer_size / sizeof(float));
+            if (JXL_DEC_SUCCESS != JxlDecoderSetImageOutBuffer(decoder.get(), &format, pixels.data(), buffer_size)) {
+                throw runtime_error{"Failed to set output buffer for JPEG XL image."};
+            }
+
+            // Read the pixel data into the pixels buffer
+            status = JxlDecoderProcessInput(decoder.get());
+
+            // Check if we got any pixels
+            if (pixels.empty()) {
+                throw runtime_error{"Failed to decode JPEG XL image data."};
+            }
+
+            result.emplace_back();
+            ImageData& data = result.back();
+
+            // Set dimensions
+            Vector2i size{(int)info.xsize, (int)info.ysize};
+            data.dataWindow = {
+                {0,        0       },
+                {size.x(), size.y()}
+            };
+            data.displayWindow = data.dataWindow;
```

Comment:
You can safely remove these as well. They'll be set to the image size automatically if unset.

At the moment, they're only really useful for EXR images, where the display region of an image can be configured separately from the region that actually has data.